### PR TITLE
Fs add prospects to campaign 61

### PR DESCRIPTION
**Pull request by mcgraths7**
**What it does**

1. Added contexts to track selected campaign, as well as selected prospects
2. Adjusted ListSelect component to be more flexible, and passed in a selectItem function which can be tied to context setter, in this case setSelectedCampaign
3a. Added button to add selected prospects to selected campaign
3b. Added prospects reference to Campaign model, and a method to add references to prospect object ids

**Known issues**

1. I tried to fix up references to ListSelect to utilize the new props, but might need to revisit in another PR if I broke any functionality
2. I still don't have an overall picture of what I'd like the app to look like in my head, so the styling is a little inconsistent. Will revisit overhauling the app style in a later issue
3. Leftmost "upload prospect" button currently has no function. Possibly can replace this with add to campaign, so that scrolling is no longer necessary

### CODE REVIEW

File: server/api/prospect_handler.py [hunk 1]
```diff
@@ -52,25 +50,26 @@ def upload_prospects():
     # Add to db
     prospect_json = prospect_json.copy()
     prospect_list = list(prospect_json['prospects'].values())
-    owner_email = prospect_json['owner']
-    owner = User.get_by_email(owner_email)
+    owner = User.get_by_email(get_jwt_identity()["email"])
     dup_prospects = 0
```
Comment (Patton-L) [inline server/api/prospect_handler.py]: The dup_prospects is not used

File: server/db/model.py [hunk 2]
```diff
@@ -41,8 +41,18 @@ class Prospect(MongoModel):
     first_name = fields.CharField()
     last_name = fields.CharField()
     status = fields.CharField()
+    campaigns = fields.ListField(fields.ReferenceField(
+        "Campaign"))
+
+    @staticmethod
+    def find_by_id(id):
+        prospect = Prospect.objects.raw({'_id': id})
```
Comment (Patton-L) [inline server/db/model.py]: Should change to `prospect = Prospect.objects.get({'_id': id})`; But this might raise NotExistError if item not found.

Comment (Patton-L) [reply server/db/model.py]: as raw will return a queryset; you can also do `list(Prospect.objects.raw({'_id': id}))` to get a list

File: server/api/prospect_handler.py (hunk 1 above)
Comment (mcgraths7) [reply server/api/prospect_handler.py]: Should be fixed in latest commit. Also removed from frontend as well

**Review by Patton-L — CHANGES_REQUESTED** [review]

File: server/db/model.py (hunk 2 above)
Comment (mcgraths7) [reply server/db/model.py]: Would wrapping in a try/except block handle that?

        try:
            prospect = Prospect.objects.get({'_id': id})
            return prospect
        except:
            return None

or would I need to specify which exception to check for

Comment (Patton-L) [reply server/db/model.py]: > Would wrapping in a try/except block handle that?
> 
> ```
>     try:
>         prospect = Prospect.objects.get({'_id': id})
>         return prospect
>     except:
>         return None
> ```
> 
> or would I need to specify which exception to check for

You can either do that(except DoesNotExist:), or just let the error raise and handle the error with app_errorhandler (see last line of campaign_handler.py)

**Review by Patton-L — APPROVED** [review]

**Pull request merged**